### DISCUSSION TITLE
Suggestion to update label property type

### DISCUSSION THREAD

**Discussion by marcus-j-davies** [zwave-js/zwave-js#2690] [Ideas]
Very often I see config files that are in support of various devices, with nothing more than a difference in there label.
Eaton being just 1 (have just raised a PR for one)

I assume, the labels may change between regions.

Currently, it seems this is managed by simply adding the various labels as one string: "Label 1 / Label 2".
Would it be more "structured" to allow this to come in the form of an array also?

`"label": "1 Label"`
`"label": ["Label 1", "Label 2"]`

This already happens with the `zwaveAllianceId` key

The driver (when seeing this as an array) will just concatenate using / as the separator - as to not break front ends, or the indexing format.

I think this makes it align with the `zwaveAllianceId` property?

@blhoward2 - not sure if you have a view on this?

**Comment by AlCalzone** [comment]
I think this is basically https://github.com/zwave-js/node-zwave-js/issues/1675 (Move manufacturer labels into devices array), but for device labels. I guess it could make sense to be able to specify the device and/or manufacturer in the devices array instead of the root level. That way we could assign device labels to individual IDs without splitting the files.

**Reply by blhoward2** [reply]
I'm not opposed to this but I also don't think we should also assume it's always the case they should go into one file. Sometimes there are differences (the Aeotec smart switch 7 for different regions has different model numbers, parameters, and firmware...and one is a 500 series chip and the other a 700 IIRC). Where this usually happens is when the manufacturer doesn't change product type and ID between models, like where Yale sells several models that all fingerprint the same.

**Reply by marcus-j-davies** [reply]
I'm probably not qualified enough, to offer my opinion down to that level.
But the likes of Jasco, Eaton, Yale + many more, have a lot of labels as one string.

Formatting (or at least, making them) a collection of some sort, I think will offer a cleaner interface, for label management.

**Reply by AlCalzone** [reply]
> I also don't think we should also assume it's always the case they should go into one file. 

I don't think this should be an incentive to do that, but if they are already, this could help.
https://github.com/zwave-js/node-zwave-js/pull/2689/files

> like where Yale sells several models that all fingerprint the same.

In this case all we can do is structure the labels as we do right now.

> Formatting (or at least, making them) a collection of some sort, I think will offer a cleaner interface, for label management.

I don't like making it an array just because we can, this just means we have to concatenate the labels again after reading the file. If we can say (like in your linked PR) that device ID 1 is label 1 and device ID 2 is label 2, with the device being identical, that's okay.

**Reply by marcus-j-davies** [reply]
Here is an interesting approach.

If a `label` appears for a set of ID's - it takes priority, if not, falls back to the root label?
This will work, for the current format, and the one depicted here https://github.com/zwave-js/node-zwave-js/issues/1675

or am I jumping ahead? - I have tendency to do that.

**Reply by blhoward2** [reply]
This seems overly complicated. If the devices are different enough they need different labels, and can be distinguished, they like have different manuals, etc. I’d rather them just be different files in that case. If they can’t be distinguished by the fingerprint then we’re just back to concatating them back together again.

**Reply by marcus-j-davies** [reply]
i.e -  jumping ahead 😄
